nwbinspector/tools.py

```diff
@@ -22,3 +26,39 @@ def get_nwbfile_path_from_internal_object(obj):
     if isinstance(obj, NWBFile):
         return obj.container_source
     return obj.get_ancestor("NWBFile").container_source
+
+
+def get_s3_urls_and_dandi_paths(dandiset_id: str, version_id: Optional[str] = None, n_jobs: int = 1) -> Dict[str, str]:
+    """
+    Collect S3 URLS from a DANDISet ID.
+
+    Returns dictionary that maps each S3 url to the displayed file path on the DANDI archive content page.
+    """
+    assert re.fullmatch(
+        pattern="^[0-9]{6}$", string=dandiset_id
+    ), "The specified 'path' is not a proper DANDISet ID. It should be a six-digit numeric identifier."
+
+    s3_urls_to_dandi_paths = dict()
+    if n_jobs != 1:
+        with DandiAPIClient() as client:
+            dandiset = client.get_dandiset(dandiset_id=dandiset_id, version_id=version_id)
+            max_workers = n_jobs if n_jobs > 0 else None
+            with ProcessPoolExecutor(max_workers=max_workers) as executor:
+                futures = []
+                for asset in dandiset.get_assets():
```

Comment:
```suggestion
                for asset in dandiset.get_assets():
                    if asset.path.split(".")[-1] == "nwb"
```

NWB datasets can contain non-NWB assets, so let's skip those